apps/web/src/components/SocketListener/SocketListener.store.js

```diff
@@ -102,20 +102,29 @@ export function ormSessionReducer (session, { meta, type, payload }) {
     case RECEIVE_POST: {
       currentUser = Me.first()
       const { post } = payload.data
+      const groupId = payload.groupId
       if (currentUser && post.creatorId !== currentUser.id) {
         const increment = obj =>
           obj && obj.update({
             newPostCount: (obj.newPostCount || 0) + 1
           })
 
-        post.topics.forEach(topicId => {
-          increment(GroupTopic.safeGet({
-            topic: topicId, group: post.groupId
-          }))
+        increment(TopicFollow.filter(tf =>
+          post.topics.some(t => t.id === tf.topic) && tf.group === groupId).first())
+
+        const group = Group.withId(groupId)
+        const contextWidgets = group.contextWidgets.items
+        const newContextWidgets = contextWidgets.map(cw => {
+          if (cw.type === 'chat' && post.topics.some(t => t.id === cw.viewChat?.id)) {
+            return { ...cw, highlightNumber: cw.highlightNumber + 1 }
+          }
+          return cw
         })
+        group.update({ contextWidgets: { items: structuredClone(newContextWidgets) } })
 
+        // TODO: is this working?
```

Comment:
??

Reply:
i havent yet tested the group level specific badges, and i dont think this is quite right